config.yaml

```diff
@@ -0,0 +1,77 @@
+credentials:
+  users:
+    client:
+      password: 'secret'
+      roles: [super]
+    replicator:
+      password: 'secret'
+      roles: [replication]
+    storage:
+      password: 'secret'
+      roles: [sharding]
+
+iproto:
+  advertise:
+    peer:
+      login: replicator
+    sharding:
+      login: storage
+
+sharding:
+  bucket_count: 3000
+
+groups:
+  storages:
+    roles: ['roles.sharded-queue-storage']
+    roles_cfg:
+      roles.sharded-queue-storage:
+        cfg:
+          metrics: true
+        tubes:
+          test_tube: {}
+    sharding:
+      roles: [storage]
+    replication:
+      failover: manual
+    replicasets: 
+      storage-001:
+        leader: storage-001-a
+        instances: 
+          storage-001-a:
+            iproto:
+              listen:
+                - uri: localhost:3302
+          storage-001-b:
+            iproto:
+              listen:
+                - uri: localhost:3303
+      storage-002:
+        leader: storage-002-a
+        instances: 
+          storage-002-a:
+            iproto:
+              listen:
+                - uri: localhost:3304
+          storage-002-b:
+            iproto:
+              listen:
+                - uri: localhost:3305
+  routers:
+    roles: ['roles.sharded-queue-router']
+    roles_cfg:
+      roles.sharded-queue-router:
+        cfg:
+          metrics: true
+        tubes:
```

Comment:
One should set tubes both on router and storage?

Reply:
Yes, it is. The router required an information about tubes to make possible to use the API:
```
queue.tube[tube_name]:some_call()
```

Reply:
It is allowed to inspect other instances configuration in newer Tarantool 3 versions: https://github.com/tarantool/tarantool/pull/9808 . This one was introduced for sharded application, so it seems that we're in the similar situation here. I think we should use it here: specify tubes list in a single place (I think it should be storages, but I'm not sure) and use it both in routers and storages (since, in case of configuration mismatch, using the API will fall apart).

Reply:
Thank you for the idea, updated.